syntax/ocaml.vim

```diff
@@ -226,11 +232,11 @@ else
   syn match    ocamlOperator   "<-"
 endif
 
-syn match    ocamlNumber        "\<-\=\d\(_\|\d\)*[l|L|n]\?\>"
-syn match    ocamlNumber        "\<-\=0[x|X]\(\x\|_\)\+[l|L|n]\?\>"
-syn match    ocamlNumber        "\<-\=0[o|O]\(\o\|_\)\+[l|L|n]\?\>"
-syn match    ocamlNumber        "\<-\=0[b|B]\([01]\|_\)\+[l|L|n]\?\>"
-syn match    ocamlFloat         "\<-\=\d\(_\|\d\)*\.\?\(_\|\d\)*\([eE][-+]\=\d\(_\|\d\)*\)\=\>"
+syn match    ocamlNumber        "-\=\<\d\(_\|\d\)*[l|L|n]\?\>"
+syn match    ocamlNumber        "-\=\<0[x|X]\(\x\|_\)\+[l|L|n]\?\>"
+syn match    ocamlNumber        "-\=\<0[o|O]\(\o\|_\)\+[l|L|n]\?\>"
+syn match    ocamlNumber        "-\=\<0[b|B]\([01]\|_\)\+[l|L|n]\?\>"
+syn match    ocamlFloat         "-\=\<\d\(_\|\d\)*\.\?\(_\|\d\)*\([eE][-+]\=\d\(_\|\d\)*\)\=\>"
```

Comment:
This doesn't look right to me. The block boundary (`\<`) before the `-` always matches, but the new version considers the second `-` in `1--4` to be part of the number.

Reply:
As far as I understand, the minus sign `-` is not a word character (as described by option `'iskeyword'`), hence it can never be the beginning of a word. So, with the old version, the minus sign was never included in the match. With the new version, it is.

If I’m not mistaken, the issue you underline is that, currently, the `--` operator is not matched as a whole. Neither are most OCaml operators, in fact. Only an ad-hoc set of operators is recognized. I have a patch in store for matching all OCaml operators accurately (I essentially took the grammar described by the OCaml manual and wrote regexes for it). I may consider submit it some day. I haven’t yet because it’s a breaking change. Would that be welcomed?

Reply:
Ah, I had a misunderstanding about how `\<` works. I still think this change should be reverted until a more precise matching of operators is available. It would certainly be welcomed.